lib/base/io-engine.hpp

```diff
@@ -163,51 +166,73 @@ class AsioConditionVariable
 /**
  * I/O timeout emulator
  *
+ * This class provides a workaround for Boost.ASIO's lack of built-in timeout support.
+ * While Boost.ASIO handles asynchronous operations, it does not natively support timeouts for these operations.
+ * This class uses a boost::asio::deadline_timer to emulate a timeout by scheduling a callback to be triggered
+ * after a specified duration, effectively adding timeout behavior where none exists.
+ * The callback is executed within the provided strand, ensuring thread-safety.
+ *
+ * The constructor returns immediately after scheduling the timeout callback. The callback itself is invoked
+ * asynchronously when the timeout occurs, but control is immediately returned to the caller without waiting for the
+ * timeout to expire. This allows the caller to continue execution while the timeout is running in the background.
```

Comment:
> **The constructor returns immediately after scheduling the timeout callback.** The callback itself is invoked asynchronously when the timeout occurs, **but control is immediately returned to the caller without waiting for the timeout to expire.**

Is there any difference in what the two highlighted parts are trying to say? I find the second highlighted part rather confusing as it says something about returning control right after it talked about invoking the callback, but this seems to talk about returning control from the constructor, not the callback.